File: frontend/api/corsi.py
"""Funzioni API per i corsi."""

from frontend.api.client import get, post


def get_corsi() -> list:
    """Restituisce i corsi visibili per l'utente corrente."""
    return get("/corsi/") or []


def crea_corso(
    codice: str, titolo: str, tipo_finanziamento: str,
    responsabile_id: int, num_partecipanti: int,
    data_inizio: str, data_fine: str, descrizione: str = None
) -> dict | None:
    """Crea un nuovo corso."""
    return post("/corsi/", data={
        "codice":              codice,
        "titolo":              titolo,
        "tipo_finanziamento":  tipo_finanziamento,
        "responsabile_id":     responsabile_id,
        "num_partecipanti":    num_partecipanti,
        "data_inizio":         data_inizio,
        "data_fine":           data_fine,
        "descrizione":         descrizione,
    })

File: frontend/src/api/corsi.js
// ─────────────────────────────────────────────────────────────────────────────
// API — Corsi
// ─────────────────────────────────────────────────────────────────────────────
// TODO: integrare getCorsi() in NuovaPrenotazionePage.vue per sostituire
//       l'input numerico corso_id con una <select> caricata da backend.
//       Uso: const corsi = await getCorsi({ attivo: true, sede_id: sedeId })

import { apiGet } from './client'

/**
 * Lista corsi con filtri opzionali.
 * @param {{ sede_id?: number, attivo?: boolean }} params
 * @returns {Promise<Array<{ id, codice, titolo, sede_id, stato_del_corso, tipo_finanziamento, data_inizio_corso, data_fine_presunta, attivo }>>}
 */
export async function getCorsi(params = {}) {
  const qs = new URLSearchParams()
  if (params.sede_id !== undefined) qs.set('sede_id', params.sede_id)
  if (params.attivo  !== undefined) qs.set('attivo',  params.attivo)
  const query = qs.toString() ? `?${qs}` : ''
  return apiGet(`/corsi/${query}`)
}

/**
 * Dettaglio completo di un corso.
 * @param {number} id
 */
export async function getCorso(id) {
  return apiGet(`/corsi/${id}`)
}